### ISSUE TITLE
ImGui::SetKeyboardFocusHere doesn't highlight selected item

### ISSUE THREAD

**Issue by Woitek1993** [ocornut/imgui#7370]
### Version/Branch of Dear ImGui:

Version 1.90.0, Branch: docking

### Back-ends:

imgui_impl_glfw.cpp + imgui_impl_opengl2.cpp

### Compiler, OS:

Windows 10

### Details:

Hello, 
I am trying to make a programmable change of the focus of the keyboard on the item using SetKeyboardFocusHere();; I've noticed that the highlight is not applied while doing so. I would like to get a highlight on this item in the same way as when the down/up navigation keys are pressed. The object in this case does not lose the highlight when the mouse is moved away.

Example code:
        
        ImGui::Selectable(std::format("Name: \"{0}\"", (t->m_name.c_str()) ? t->m_name.c_str() : "").c_str());
        if (ImGui::IsItemClicked())
        {
            SetKeyboardFocusHere();
        }
        ImGui::Selectable(std::format("Mask Name: \"{0}\"", (t->m_mask_name.c_str()) ? t->m_mask_name.c_str() : "").c_str());
        if (ImGui::IsItemClicked())
        {
            SetKeyboardFocusHere();
        }
        ImGui::Selectable(std::format("Width: {0}", t->m_width).c_str());
        if (ImGui::IsItemClicked())
        {
            SetKeyboardFocusHere();
        }

### Screenshots/Video:

Got:
![got](https://github.com/ocornut/imgui/assets/26350423/c635ed01-02de-4423-b2ec-0e38d63de98d)
Expected:
![expected](https://github.com/ocornut/imgui/assets/26350423/5a5752ee-bbf2-42eb-bbb0-3694d57b17c8)

**Comment by PathogenDavid** [comment]
> I've noticed that the highlight is not applied while doing so. I would like to get a highlight on this item in the same way as when the down/up navigation keys are pressed.

The nav highlight is intentionally hidden when the mouse is used since the highlight is only intended only for keyboard and gamepad navigation, so you're fighting against the internal design of the nav system.

It would be helpful to know exactly what you're trying to accomplish UI-wise from a user perspective as this feels very [XY problem](https://xyproblem.info/).

Are you wanting the nav highlight to always be visible when the user interacts with all items or just for these selectables?

If the latter, why? And could the built-in selectable highlighting suitable for your use-case?

---------------

As an aside, `ImGui::SetKeyboardFocusHere();` without any arguments focuses the *next* item, you need to use `ImGui::SetKeyboardFocusHere(-1);`

**Comment by ocornut** [comment]
I have added a few things related to those requests:

- Nav: added `io.ConfigNavCursorVisibleAuto` and `io.ConfigNavCursorVisibleAlways` config options (ab9ce2a).
- Nav: added `SetNavCursorVisible()` to override current state (634a7ed9).

Technically I think the config options are sufficient for most people.

Other tangentially related options:
- Nav: added `io.ConfigNavEscapeClearFocusItem` (7a56b41).
- Nav: added `io.ConfigNavEscapeClearFocusWindow` (b001038).
- And a couple of related fixes for all the above.

Technically this # also pertain to use SetKeyboardFocusHere() and other functions which are still be redesigned/improved.
**My expectation is that the future replacement `SetKeyboardFocusHere()` will have options to configure how nav cursor visibility is made visible or not.**

I'm closing this and other related issues as normally solved, but feel free to comment or open new issues if you have question or problem related to those.

(

Also note, if you have mods using internals, that those two fields have been renamed internally:
- Terminology has been changed to call this the "nav cursor" more consistently.
- bool NavDisableHighlight  -> became bool NavCursorVisible (with opposite value). 
- bool NavDisableMouseHover -> became bool NavDisableMouseHover

)
